src/transformStateWithClones.js

```diff
@@ -7,7 +7,41 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copyState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    transform(copyState, action);
+    result.push({ ...copyState });
+  }
+
+  return result;
+}
+
+function transform(copyState, action) {
+  switch (action.type) {
+    case 'addProperties':
```

Comment:
It would be cool to put the option names in separate constants.

Reply:
U mean, make constants as sting values for different cases?